dask/dataframe/io/csv.py

```diff
@@ -70,6 +71,8 @@ def project_columns(self, columns):
         if columns == self.columns:
             return self
         func = copy.deepcopy(self)
+        func.head = self.head[columns]
+        func.dtypes = {c: self.dtypes[c] for c in columns}
```

Comment:
Do we need to something similar for other `project_columns` implementation? For example, modifying the `dtype` in `MakeTimeseriesPart.project_columns`?

Reply:
Good point.  We have been updating `dtypes` in `MakeTimeseriesPart.__call__`, rather than in `MakeTimeseriesPart.project_columns` (which is correct, but pretty silly) -- I just pushed a fix for this.